src/antlr/Parser.g

```diff
@@ -1832,10 +1831,9 @@ shorthandColumnOperation[UpdateStatement.OperationCollector operations, ColumnId
       {
             if (isParsingTxn)
             {
-                ReferenceValue.Raw left = new ReferenceValue.SelfReference(key);
                 ReferenceValue.Raw right = new ReferenceValue.Constant.Raw(t);
-                ReferenceValue.Raw operation = $sig.text.equals("+=") ? new ReferenceValue.Addition.Raw(left, right) : new ReferenceValue.Subtraction.Raw(left, right);
-                addRawReferenceOperation(operations, key, new ReferenceOperation.Assignment.Raw(key, operation));
+                Operation.RawUpdate operation = $sig.text.equals("+=") ? new Operation.Addition(t) : new Operation.Substraction(t);
+                addRawReferenceOperation(operations, key, new ReferenceOperation.Assignment.Raw(operation, key, right));
```

Comment:
I think we can remove this as we now can figure out if we need to run before or after reads